airbyte-integrations/connectors/source-stripe/source_stripe/source.py

```diff
@@ -38,15 +40,11 @@
 
 
 class SourceStripe(AbstractSource):
-    def __init__(self, catalog_path: Optional[str] = None, **kwargs):
+    def __init__(self, catalog: ConfiguredAirbyteCatalog, **kwargs):
```

Comment:
It was like this before but this interface feels weird as I would expect:

```
source = SourceStripe(<a catalog with only one stream>
streams = source.streams()
assert len(streams) == 1
```

However, all streams are returned there. I guess otherwise would be a breaking change as streams has always returned everything. There are also cases where we don't have the catalog (for example, `discover`)